File: nexus-backend/src/main/java/tn/arctic/nexus/entities/ChatMessage.java
package tn.arctic.nexus.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import org.jetbrains.annotations.NotNull;


import java.time.LocalDateTime;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessage {


    private Long id;
    private String sendername;
    private Long senderId;
    private Long recipientId;
    private String content;
    private MessageType type;
    private LocalDateTime timestamp = LocalDateTime.now();

}

File: nexus-frontend/src/app/core/entities/user/ChatMessage.ts
export class ChatMessage {
    id: number;
    senderId: number;       // ID de l'expéditeur, type number
    recipientId: number;    // ID du destinataire, type number
    content: string;
    timestamp: Date;

    constructor(senderId: number, recipientId: number, content: string, id: number = 0) {
        this.senderId = senderId;
        this.recipientId = recipientId;
        this.content = content;
        this.timestamp = new Date();  // Ajoute la date et l'heure actuelles
        this.id = id; // Initialisation de l'ID
    }
}
